libfwupdplugin/fu-bios-settings.c

```diff
@@ -174,27 +176,15 @@ fu_bios_setting_set_current_value(FwupdBiosSetting *attr, GError **error)
 #define LENOVO_EXCLUDED		"[Excluded from boot order:"
 
 static void
-fu_bios_setting_fixup_read_only(FwupdBiosSetting *attr)
+fu_bios_setting_set_read_only(FuBiosSettings *self, FwupdBiosSetting *attr)
 {
 	if (fwupd_bios_setting_get_kind(attr) == FWUPD_BIOS_SETTING_KIND_ENUMERATION) {
-		struct {
-			const gchar *id;
-			const gchar *value;
-		} read_only_map[] = {{"com.thinklmi.SecureBoot", "Enable"},
-				     {"com.dell-wmi-sysman.SecureBoot", "Enabled"},
-				     {NULL, NULL}};
 		const gchar *id = fwupd_bios_setting_get_id(attr);
-		const gchar *tmp = fwupd_bios_setting_get_current_value(attr);
-		for (guint i = 0; read_only_map[i].id != NULL; i++) {
-			if (g_strcmp0(id, read_only_map[i].id) != 0)
-				continue;
-
-			if (read_only_map[i].value == NULL ||
-			    g_strcmp0(tmp, read_only_map[i].value) == 0) {
-				fwupd_bios_setting_set_read_only(attr, TRUE);
-				return;
-			}
-		}
+		if (!g_hash_table_contains(self->read_only, id))
+			return;
+		if (g_strcmp0(g_hash_table_lookup(self->read_only, id),
```

Comment:
seems inefficient to look this up twice; just do `const gchar *value = g_hash_table_lookup(self->read_only, id)` then check `value` for NULL.